### ISSUE TITLE
feature request: access to raw git_diff_stats

### ISSUE THREAD

**Issue by josharian** [libgit2/libgit2#5231]
I would like to print diff stats, but in a different format than those already supported by git_diff_stats_to_buf. (As a motivating example similar to mine, consider adding color.)

To that end, I'd like to have some API so that I can get access to the components of the git_diff_stats.

Thanks!

**Comment by josharian** [comment]
If this is OK, and we can agree on an API, I'll take a stab at implementing. I can also propose an API if that'd help.

**Comment by ethomson** [comment]
I think this is a reasonable idea.  I think an API proposal would be a good first step - it feels like that struct shouldn't be exposed as-is, so maybe some refactoring or some other accessor function on top would be reasonable.

**Comment by josharian** [comment]
I spent some quality time with this today and came to the conclusion that the amount of new API involved isn't obviously worth it. Most of the interesting code is around display considerations (calculating max name lengths, etc.), but the whole purpose of the API is to allow alternative displays.

I've concluded that it is probably better to have the caller just use `git_diff_num_deltas`, `git_patch_from_diff`, `git_patch_line_stats`, and friends and do the glueing themselves.

I'll close this now; apologies for the noise.
